src/compiler/diagnosticMessages.json

```
@@ -936,7 +936,7 @@
         "category": "Error",
         "code": 1323
     },
-    "Dynamic import must have one specifier as an argument.": {
+    "The second argument of an import call is only supported when the module flag is not available in the current module flag.": {

```

The wording here needs some work, I think.

Perhaps this would work better:
```suggestion
    "Dynamic import only supports a second argument when the '--module' option is set to 'esnext'.": {
```

/CC @DanielRosenwasser